### PR TITLE
[Distributed] Strip marker protocols from distributed thunks and accessible functions

### PR DESCRIPTION
**Pull request by xedin**
Since marker protocols do not exist at runtime, let's strip them from mangling of distributed thunks
and generic environments associated with accessible functions.

Resolves: rdar://90293494

### CODE REVIEW

File: lib/AST/GenericSignature.cpp [hunk 1]
```diff
@@ -1163,3 +1163,24 @@ swift::buildGenericSignature(ASTContext &ctx,
...
+
+  // Drop all conformance requirements to marker protocols (if any).
+  llvm::copy_if(requirements, std::back_inserter(reducedRequirements),
+                [](const Requirement &requirement) {
+                  if (requirement.getKind() == RequirementKind::Conformance) {
+                    auto *protocol = requirement.getProtocolDecl();
+                    return !protocol->isMarkerProtocol();
+                  }
+                  return true;
+                });
```
Comment (ktoso) [inline lib/AST/GenericSignature.cpp:1179]: I see, nice

**Review by ktoso — APPROVED** [review]
Nice,:l thank you!

**Comment by ktoso** [comment]
@swift-ci please smoke test

**Comment by xedin** [comment]
@swift-ci please test macOS platform

**Comment by xedin** [comment]
Let's run a full suite just in case.

**Comment by ktoso** [comment]
Yeah, sounds good 👍

**Comment by ktoso** [comment]
CI node died, starting again

```
uest: Termination requested at Fri Mar 18 05:34:12 UTC 2022 by Thread[Ping thread for channel hudson.remoting.Channel@1909955:macos-node-i-0c48462df7e9f71e1,5,main] [id=2687760]
		at hudson.model.Computer.recordTermination(Computer.java:232)
		at hudson.model.Computer.disconnect(Computer.java:500)
```

@swift-ci please test macOS platform

**Comment by ktoso** [comment]
and full test runs all good 🥳

**Pull request merged**